### DISCUSSION TITLE
[FEAT] Support for 'filter' bulkSelectMode

### DISCUSSION THREAD

**Discussion by davidRamirez90** [shlomiassaf/ngrid#76] [Q&A]
#### Please describe the feature you would like to request.

Allow for an extra option for `bulkSelectMode` of the Selection-Column plugin in addition to 'view', 'all' and 'none', namely 'filter'.

#### What is the use-case or motivation for this proposal?

When applying filters to the dataSource, the selection model saves the filtered data under `ds.filteredData`.  It would be a nice option to allow the 'select all' checkbox to select only the data that has been filtered.

**Comment by shlomiassaf** [comment]
The datasource has a selection property that reflects the selected items.

You can track the filtered data and match the selection to it... This doesnt require a special directive
